Source/WebCore/page/PerformanceObserver.cpp

```diff
@@ -93,11 +91,9 @@ ExceptionOr<void> PerformanceObserver::observe(Init&& init)
             isBuffered = true;
             auto oldSize = m_entriesToDeliver.size();
             protectedPerformance()->appendBufferedEntriesByType(*init.type, m_entriesToDeliver, *this);
-            auto begin = m_entriesToDeliver.begin();
-            auto oldEnd = begin + oldSize;
-            auto end = m_entriesToDeliver.end();
-            std::stable_sort(oldEnd, end, PerformanceEntry::startTimeCompareLessThan);
-            std::inplace_merge(begin, oldEnd, end, PerformanceEntry::startTimeCompareLessThan);
+            auto entriesToDeliver = m_entriesToDeliver.mutableSpan();
+            std::stable_sort(entriesToDeliver.subspan(oldSize).begin(), entriesToDeliver.end(), PerformanceEntry::startTimeCompareLessThan);
+            std::inplace_merge(entriesToDeliver.begin(), entriesToDeliver.subspan(oldSize).begin(), entriesToDeliver.end(), PerformanceEntry::startTimeCompareLessThan);
```

Comment:
> I would be surprised if span iterator operations were unchecked.

Then be surprised :) cc @geoffreygaren.
We may need to create our own WTF::Span so we get checked iterators, libcpp couldn't do it due to ABI compatibility iirc.

Reply:
Vector iterators are currently raw pointers and not safe either. We'll likely want to make them safe as well though, I agree.

Reply:
I don’t want to stand in the way, but I am not clear on the value of the current change. If `Vector` or `span` iterators were checked then this code would be significantly safer. If not, making just the `subspan` call do a range check doesn’t seem all that valuable, and it’s a significant code change. I suppose we need to do something to make the validation checker happy, but it seems a smaller change could possibly accomplish that?

Reply:
Unfortunately, libc++ span iterators are in fact unchecked in user space, and there's no plan to fix that for the time being.

Making Vector iterator checked is an option. (And we should totally do that, or at minimum mark Vector iterator ++ WTF_UNSAFE_BUFFER_USAGE as soon as possible.)

Making a checked span iterator wrapper is also an option. Note that we don't necessarily need to replace all use of span with a new WTF::Span. We can just use a discipline of doing boundsCheckedBegin(span) instead of span.begin(), and fill out more features in BoundsCheckedIterator as we need them. And maybe pick a shorter name while we're at it.

(libc++ is working on adding [[clang::unsafe_buffer_usage]] to span iterators in user space, so we do have a solution to using a span iterator by accident.)

Reply:
I guess the main thing I am wondering about is how we align the unsafe checks with the strategy. I’m surprised that we can make the checker happy with these changes.

Reply:
If we have a checker that can check when it’s needed, than I think `saferBegin(x)` would be fine. For both `Vector` and `span`.